text/0000-partial_types.md

```diff
@@ -0,0 +1,672 @@
+- Feature Name: `partial_types`
+- Start Date: 2023-04-18
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+
+# Summary
+[summary]: #summary
+
+Partial types proposal is a generalization on "partial borrowing"-like proposals (more correct name is "partial not borrowing" or "qualified borrowing" since Rust allows partial borrowing already).
+
+This proposal is a universal roadmap "how to do partial not consumption (including partial not borrowing) right", and not under the hood of the Rust compiler.
+
+Partial Types is a **minimal** and full extension to the Rust Type System, which allows to safe control easily partial parameters and all kinds of partial not consumption.
+
+Advantages: maximum type safety, maximum type control guarantee, no ambiguities, flexibility, usability and universality.
+
+
+# Motivation
+[motivation]: #motivation
+
+Safe, Flexible controllable partial parameters for functions and partial not consumption (including partial not borrowing) are highly needed and this feature unlock huge amount of possibilities.
+
+Partial borrowing is already possible in Rust, as partial referencing and partial moves.
+
+But partial parameters are forbidden now, as qualified consumption: partial not borrowing, partial not referencing, partial not moving and partial initializing.
+
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+_**Note**: I didn't comment type explanations (as is needed for compiler) for colorizing purposes only._
+
+_**Note**: I use symbol `~` in code as a synonym of "equivalent" word only. It is not a Rust operator!_
```

Comment:
This is very confusing to me to read throughout this section. Although it would be more verbose, I would much prefer if you just explained things in comments rather than putting it directly in the code, since it begs to be read directly as syntax regardless of what you say.

Reply:
Ok, I did it: I commented type explanations.
Thanks for advice.